source/interface/transport_interface.h

```diff
@@ -183,7 +183,10 @@ typedef int32_t ( * TransportRecv_t )( NetworkContext_t * pNetworkContext,
  * @param[in] pBuffer Buffer containing the bytes to send over the network stack.
  * @param[in] bytesToSend Number of bytes to send over the network.
  *
- * @return The number of bytes sent or a negative error code.
+ * @return The number of bytes sent or a negative value to indicate error.
+ * If no data is transmitted over the network due to a full TX buffer and
```

Comment:
Nit: We used `@note`here when making a similar comment for TransportRecv_t. Also I think there would have to be an empty line between the `@return` and `@note` blocks, judging from the other doxygen comments

Reply:
Made the API doc for send and receive consistent.